msal/src/main/java/com/microsoft/identity/client/statemachine/results/ResetPasswordResult.kt

```diff
@@ -0,0 +1,178 @@
+     *
+     * @param nextState [com.microsoft.identity.client.statemachine.states.ResetPasswordCodeRequiredState] the current state of the flow with follow-on methods.
+     * @param codeLength the length of the code required by the server.
+     * @param sentTo the email/phone number the code was sent to.
+     * @param channel the channel(email/phone) the code was sent through.
+     */
+    class CodeRequired(
+        override val nextState: ResetPasswordCodeRequiredState,
+        val codeLength: Int,
+        val sentTo: String,
+        val channel: String,
+    ) : ResetPasswordStartResult, Result.SuccessResult(nextState = nextState)
+
+    /**
+     * UserNotFound ErrorResult, which indicates there was no account found with the provided email.
+     * The flow should be restarted.
+     *
+     * @param error [com.microsoft.identity.client.statemachine.UserNotFoundError] the current state of the flow with follow-on methods.
+     */
+    class UserNotFound(
+        override val error: UserNotFoundError
+    ) : ResetPasswordStartResult, Result.ErrorResult(error = error)
+}
+
+/**
+ * SSPR submit code result, produced by
+ * [com.microsoft.identity.client.statemachine.states.ResetPasswordCodeRequiredState.submitCode]
+ */
+sealed interface ResetPasswordSubmitCodeResult : Result {
+    /**
+     * PasswordRequired Result, which indicates that a valid new password is required from the user to continue.
+     *
+     * @param nextState [com.microsoft.identity.client.statemachine.states.ResetPasswordPasswordRequiredState] the current state of the flow with follow-on methods.
+     */
+    class PasswordRequired(
+        override val nextState: ResetPasswordPasswordRequiredState
+    ) : ResetPasswordSubmitCodeResult, Result.SuccessResult(nextState = nextState)
+
+    /**
+     * CodeIncorrect ErrorResult, which indicates the verification code provided by user is incorrect.
+     * The code should be re-submitted.
+     *
+     * @param error [com.microsoft.identity.client.statemachine.IncorrectCodeError]
+     */
+    class CodeIncorrect(
+        override val error: IncorrectCodeError
+    ) : ResetPasswordSubmitCodeResult, Result.ErrorResult(error = error)
+}
+
+/**
+ * Sign in resend code result, produced by
+ * [com.microsoft.identity.client.statemachine.states.ResetPasswordCodeRequiredState.resendCode]
+ */
+sealed interface ResetPasswordResendCodeResult : Result {
+    /**
+     * Success Result, which indicates a new verification code was successfully resent.
+     *
+     * @param nextState [com.microsoft.identity.client.statemachine.states.ResetPasswordCodeRequiredState] the current state of the flow with follow-on methods.
+     * @param codeLength the length of the code required by the server.
+     * @param sentTo the email/phone number the code was sent to.
+     * @param channel channel(email/phone) the code was sent through.
+     */
+    class Success(
```

Comment:
So you can only get success in this operation? (I don't see any other result types here)

Reply:
The other result types in the base class `Result`.